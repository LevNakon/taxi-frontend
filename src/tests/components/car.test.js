import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Car from '../../components/car';

describe('Car --- test render correctly component', () => {
    const initialState = {
        carState: {
            message: '',
            success: true,
            car: {
                brand: 'test brand',
                model: 'test model',
                year: 2000,
                run: 20000,
            }
        }
    };
    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(<Car store={store} />);
    });

    it('render correctly component Car', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('check props matches with initialState', () => {
        expect(wrapper.dive().prop('success')).toEqual(initialState.carState.success);
        expect(wrapper.dive().prop('message')).toEqual(initialState.carState.message);
        expect(wrapper.dive().prop('car')).toEqual(initialState.carState.car);
    });
});