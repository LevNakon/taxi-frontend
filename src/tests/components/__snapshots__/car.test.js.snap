// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car --- test render correctly component render correctly component Car 1`] = `
<ContextProvider
  value={null}
>
  <withRouter(Car)
    car={
      Object {
        "brand": "test brand",
        "model": "test model",
        "run": 20000,
        "year": 2000,
      }
    }
    carCreateNull={[Function]}
    carUpdateWatcher={[Function]}
    message=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    success={true}
    userGetWatcher={[Function]}
  />
</ContextProvider>
`;
